The "101-lazy_matrix_mul.py" Module
===============================
Using "lazy_matric_mul"
------------------

Import function from module
>>> lazy_matrix_mul = __import__("101-lazy_matrix_mul").lazy_matrix_mul

1. When called without arguments
>>> lazy_matrix_mul()
Traceback (most recent call last):
TypeError: lazy_matrix_mul() missing 2 required positional arguments: 'm_a' and 'm_b'

2. When called with non list argument as m_a
>>> lazy_matrix_mul(4, [[1, 2, 3]])
Traceback (most recent call last):
ValueError: Scalar operands are not allowed, use '*' instead

3. When called with non list argument as m_b
>>> lazy_matrix_mul([[1, 2, 3]], 4)
Traceback (most recent call last):
ValueError: Scalar operands are not allowed, use '*' instead

4. When called with a non list of lists argument m_a
>>> lazy_matrix_mul([1, 2, 3], [[1, 2]])
Traceback (most recent call last):
ValueError: shapes (3,) and (1,2) not aligned: 3 (dim 0) != 1 (dim 0)

5. When called with a non list of lists argument m_b
>>> lazy_matrix_mul([[1, 2, 3]], [1, 2])
Traceback (most recent call last):
ValueError: shapes (1,3) and (2,) not aligned: 3 (dim 1) != 2 (dim 0)

6. When called with an empty matrix as m_a
>>> lazy_matrix_mul([], [[1,2]])
Traceback (most recent call last):
ValueError: shapes (0,) and (1,2) not aligned: 0 (dim 0) != 1 (dim 0)

7. When called with an empty matrix as m_b
>>> lazy_matrix_mul([[1,2]], [])
Traceback (most recent call last):
ValueError: shapes (1,2) and (0,) not aligned: 2 (dim 1) != 0 (dim 0)

8. When called with a matrix containing non integer or float value as m_a
>>> lazy_matrix_mul([[1, 2, 'a']], [[1],[2]])
Traceback (most recent call last):
ValueError: shapes (1,3) and (2,1) not aligned: 3 (dim 1) != 2 (dim 0)

9. When called with matrix containing non integer or float value as m_b
>>> lazy_matrix_mul([[1, 2], [3, 4]], [[1, 'b']])
Traceback (most recent call last):
ValueError: shapes (2,2) and (1,2) not aligned: 2 (dim 1) != 1 (dim 0)

10. When called with mismatching row lengths as m_a
>>> lazy_matrix_mul([[1, 2], [2, 3, 4]], [[1, 2]])
Traceback (most recent call last):
TypeError: Object arrays are not currently supported

11. When called with mismatching row lengths as m_b
>>> lazy_matrix_mul([[1, 2]], [[1, 2], [2, 3, 4]])
Traceback (most recent call last):
TypeError: Object arrays are not currently supported

11.5. When called with mismatching row lengths as m_b
>>> lazy_matrix_mul([[5, 6, 1], [7, 8]], [[5, 6, 1], [7, 8]])
Traceback (most recent call last):
TypeError: Object arrays are not currently supported

12. When called with matrix values that cannot be multiplied
>>> lazy_matrix_mul([[1, 2, 3]], [[1, 2]])
Traceback (most recent call last):
ValueError: shapes (1,3) and (1,2) not aligned: 3 (dim 1) != 1 (dim 0)

13. When called with multipliable matrices
>>> lazy_matrix_mul([[1, 2], [3, 4]], [[1, 2], [3, 4]]) #doctest: +NORMALIZE_WHITESPACE
array([[ 7, 10],
      [15, 22]])